### DISCUSSION TITLE
Using status codes to keep particles from getting stuck on the seafloor

### DISCUSSION THREAD

**Discussion by jwongala** [Parcels-code/Parcels#1579] [Q&A]
Hello! 

I would like to use [status codes](https://docs.oceanparcels.org/en/latest/examples/tutorial_kernelloop.html#Working-with-Status-Codes) to unstick particles from the seafloor. I know there is an option to use status codes to keep particles in the ocean and move one degree west in the status codes examples, but I wanted to know if someone has done this regarding the seafloor. I have a grid of depth layers for the ROMS I am using that I have added to the `fieldset`, but I am unsure how I would reference this to unstick particles from the seafloor. Also side note: the ROMS I am using are nested fields, and the depth grids are nested and added to the fieldset. 

Thank you!

**Comment by erikvansebille** [comment]
Hi @jwongala, this is a very challenging problem. Is your model data in sigma-levels? Or in z-levels?

We don't (yet) have a good way to move particles away from the bottom, and certainly no statuscodes to deal with this. See also the discussion (about MOM, so not ROMS) in https://github.com/OceanParcels/parcels/issues/876. I don't have a simple solution for you, I fear. You'll have to write your own Kernel for this, perhaps others (@cpierard?) have some tips?

**Comment by jwongala** [comment]
Hi @erikvansebille thanks for the information. The ROMS I am working with is in z-levels. If anyone has tips I would greatly appreciate it.
